pkg/cmd/search/repos/repos.go

```diff
@@ -0,0 +1,198 @@
+    `),
+		Example: heredoc.Doc(`
+      # search repositories matching set of keywords "cli" and "golang"
+      $ gh search repos cli golang
+
+      # search repositories matching phrase "cli golang"
+      $ gh search repos "cli golang"
+
+      # search repositories in the cli organization
+      $ gh search repos --org=cli
+
+      # search repositories with a set of topics
+      $ gh search repos --topic="cli, terminal"
+
+      # search repositories by coding language and number of good first issues
+      $ gh search repos --language=golang --good-first-issues=">=10"
+
+      # search repositories by keyword "cli", number of forks, and number of stars
+      # using GitHub search syntax
+      $ gh search repos cli forks:"<100" stars:">=1000"
+    `),
+		RunE: func(c *cobra.Command, args []string) error {
+			if len(args) == 0 && c.Flags().NFlag() == 0 {
+				return cmdutil.FlagErrorf("specify search keywords or flags")
+			}
+			opts.Query.Keywords = args
+			if opts.Query.Limit < 1 || opts.Query.Limit > 1000 {
+				return cmdutil.FlagErrorf("`--limit` must be between 1 and 1000")
+			}
+			if runF != nil {
+				return runF(opts)
+			}
+			return reposRun(opts)
+		},
+	}
+
+	// Output flags
+	cmdutil.AddJSONFlags(cmd, &opts.Exporter, search.RepositoryFields)
+	cmd.Flags().BoolVarP(&opts.WebMode, "web", "w", false, "Open the search query in the web browser")
+
+	// Query parameter flags
+	cmd.Flags().IntVarP(&opts.Query.Limit, "limit", "L", 30, "Maximum number of repositories to fetch")
+	cmdutil.StringEnumFlag(cmd, &opts.Query.Order, "order", "", "", []string{"asc", "desc"}, "Order of repositories returned, ignored unless '--sort' flag is specified")
+	cmdutil.StringEnumFlag(cmd, &opts.Query.Sort, "sort", "", "", []string{"forks", "help-wanted-issues", "stars", "updated"}, "Sort fetched repositories")
+
+	// Query qualifier flags
+	cmdutil.NilBoolFlag(cmd, &opts.Query.Qualifiers.Archived, "archived", "", "Filter based on archive state")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Created, "created", "", "Filter based on created at date")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Followers, "followers", "", "Filter based on number of followers")
+	cmdutil.StringEnumFlag(cmd, &opts.Query.Qualifiers.Fork, "include-forks", "", "", []string{"false", "true", "only"}, "Include forks in fetched repositories")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Forks, "forks", "", "Filter on number of forks")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.GoodFirstIssues, "good-first-issues", "", "Filter on number of issues with the 'good first issue' label")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.HelpWantedIssues, "help-wanted-issues", "", "Filter on number of issues with the 'help wanted' label")
+	cmdutil.StringSliceEnumFlag(cmd, &opts.Query.Qualifiers.In, "in", "", nil, []string{"name", "description", "readme"}, "Restrict search to specific part of repository")
```

Comment:
💅 Like the ES `is:` qualifier, I'm not a fan of the `in:` qualifier because on its own it doesn't communicate much. For that reason, we could consider using another flag name than `--in`. How about `--match name`, `--match-field readme` or `--match-in description`?

Reply:
I agree, I think `--match` is a good alternative.